### ISSUE TITLE
corrupted output dir with addPassthroughCopy while --serve --incremental running

### ISSUE THREAD

**Issue by Gr0t3Man** [11ty/eleventy#2182]
I have a ‘assest’ directory with multiple sub folders, which are configured with a addPassthroughCopy command;

  eleventyConfig.addPassthroughCopy({ "assets/styles": "styles" });
  eleventyConfig.addPassthroughCopy({ "assets/scripts": "scripts" });
  eleventyConfig.addPassthroughCopy({ "assets/root": "/" });
  eleventyConfig.addPassthroughCopy({ "assets/images": "images" });

When eleventy is running in –-serve --incremental mode and a jpg file is added to the ‘assets/images’ dir, eleventy sees that the file is added.

<img width="837" alt="Schermafbeelding 2022-01-20 om 10 47 05" src="https://user-images.githubusercontent.com/42387489/150314493-d8c91a70-ea5e-4ecc-a4e9-46e4e9c8a8d2.png">

In the _site directory a file with the name ‘images’ is created!!?? The preview of this file shows the actual picture which was added. The directory  _site/images isn’t available anymore.

<img width="840" alt="Schermafbeelding 2022-01-20 om 10 47 23" src="https://user-images.githubusercontent.com/42387489/150314560-54c9b8e6-3791-4bef-aa3b-d3f35317b87f.png">

After this i must stop eleventy, delete the corrupted file and restart eleventy to regenerate the missing directory.

Currently I’am running v1.0.0, but I saw this problem first in the last days of December, when I was running the latetst beta version of that time.

When only running the --serve mode, the files are nicely added/changed in the output directory. 

Note: when I edit/add a file in the ‘assets/root’ folder, the entire _site directory is corrupted!!

**Environment:**

- MacOS Mojav 10.14.6
- Eleventy Version v.1.0.0

**Additional context**

**Comment by bennypowers** [comment]
Same here.

```sh
eleventy --serve --incremental --config=.eleventy.cjs
```

```js
eleventyConfig.addPassthroughCopy('docs/bundle.{js,map,ts}');
eleventyConfig.addPassthroughCopy('docs/demo.{js,map,ts}');
eleventyConfig.addPassthroughCopy('docs/main.mjs');
eleventyConfig.addPassthroughCopy('brand/**/*');

const repoRoot = join(__dirname, '..', '..');

// Copy all component and core files to _site
const files = Object.fromEntries([
  ...fs.readdirSync(join(repoRoot, 'elements')).map(dir => [
    `elements/${dir}`,
    `components/${dir.replace('pfe-', '')}`,
  ]),
  ...fs.readdirSync(join(repoRoot, 'core')).map(dir => [
    `core/${dir}`,
    `core/${dir.replace('pfe-', '')}`,
  ]),
]);

eleventyConfig.addPassthroughCopy(files);
```

The `files` object ends up looking like this:
```js
{
  'elements/pfe-accordion': 'components/accordion',
  'elements/pfe-autocomplete': 'components/autocomplete',
  'elements/pfe-etc': 'components/etc',

  'core/pfe-core': 'core/core',
  'core/pfe-sass': 'core/sass',
  'core/pfe-styles': 'core/styles'
}
```

First run is fine, `_site/core/core` is a directory full of files.

Then, if I save `core/some/thing.ts`, after the build, I notice that `_site/core/core` is no longer a directory, but a file, and it's text contents are identical to `core/some/thing.ts`

**Comment by binyamin** [comment]
Confirmed on linux.

**Comment by Gr0t3Man** [comment]
Solved in with Eleventy 1.0.1 and 2.0.0-canary.4. 
See https://github.com/11ty/eleventy/issues/2278

**Comment by stormitornado** [comment]
it's still has such bug (version 2.0.1)
I wrote this to copy everything from public to the root, at first time it copied properly with the same folder structure, but on update this structure brokes, files go directly to the root.
`eleventyConfig.addPassthroughCopy({"public": "/"});`
Maybe I should write another rule?
